periodic sync upstream KF to midstream ODH

Al-Pragliola opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- [UI] Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

- keep kubeflow/model-registry#1171
- keep kubeflow/model-registry#1172
- keep kubeflow/model-registry#1173
- keep kubeflow/model-registry#1174
- keep kubeflow/model-registry#1177
- keep kubeflow/model-registry#1168
- keep kubeflow/model-registry#1170
- keep kubeflow/model-registry#1008
- keep kubeflow/model-registry#1188
- keep kubeflow/model-registry#1189
- keep kubeflow/model-registry#1191
- (merge commit)

labels are applied,

superseeds https://github.com/opendatahub-io/model-registry/pull/221

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [ ] The commits have meaningful messages
- [ ] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](../OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

<!--- You can ignore these if you are doing Go Model Registry REST server, MR Python client, manifest, controller, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

tarilabs commented:
> we can fix that with https://github.com/opendatahub-io/model-registry/issues/211

+1 (already approved)

dependabot[bot] committed:
build(deps): bump huggingface-hub from 0.32.1 to 0.32.3 in /clients/python (#1171)

Bumps [huggingface-hub](https://github.com/huggingface/huggingface_hub) from 0.32.1 to 0.32.3.
- [Release notes](https://github.com/huggingface/huggingface_hub/releases)
- [Commits](https://github.com/huggingface/huggingface_hub/compare/v0.32.1...v0.32.3)

---
updated-dependencies:
- dependency-name: huggingface-hub
  dependency-version: 0.32.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump mypy from 1.15.0 to 1.16.0 in /clients/python (#1172)

Bumps [mypy](https://github.com/python/mypy) from 1.15.0 to 1.16.0.
- [Changelog](https://github.com/python/mypy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/python/mypy/compare/v1.15.0...v1.16.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-version: 1.16.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump boto3 from 1.38.23 to 1.38.27 in /clients/python (#1173)

Bumps [boto3](https://github.com/boto/boto3) from 1.38.23 to 1.38.27.
- [Release notes](https://github.com/boto/boto3/releases)
- [Commits](https://github.com/boto/boto3/compare/1.38.23...1.38.27)

---
updated-dependencies:
- dependency-name: boto3
  dependency-version: 1.38.27
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump ruff from 0.11.11 to 0.11.12 in /clients/python (#1174)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.11.11 to 0.11.12.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.11...0.11.12)

---
updated-dependencies:
- dependency-name: ruff
  dependency-version: 0.11.12
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aiohttp from 3.11.18 to 3.12.6 in /clients/python (#1177)

---
updated-dependencies:
- dependency-name: aiohttp
  dependency-version: 3.12.6
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Crazyglue committed:
Add initial async job configuration validation (#1168)

* feat: add initial s3-based validation for configuration using gila

Signed-off-by: Eric Dobroveanu <edobrove@redhat.com>

* chore: add todo for ID-based parameters

Co-authored-by: Matteo Mortari <matteo.mortari@gmail.com>
Signed-off-by: Eric Dobroveanu <Crazyglue@users.noreply.github.com>

* chore: clarify todo for connection to MR

Co-authored-by: Matteo Mortari <matteo.mortari@gmail.com>
Signed-off-by: Eric Dobroveanu <Crazyglue@users.noreply.github.com>

---------

Signed-off-by: Eric Dobroveanu <edobrove@redhat.com>
Signed-off-by: Eric Dobroveanu <Crazyglue@users.noreply.github.com>
Co-authored-by: Matteo Mortari <matteo.mortari@gmail.com>

dependabot[bot] committed:
build(deps): bump @patternfly/react-table from 6.2.0 to 6.2.2 in /clients/ui/frontend (#1170)

Bumps [@patternfly/react-table](https://github.com/patternfly/patternfly-react) from 6.2.0 to 6.2.2.
- [Release notes](https://github.com/patternfly/patternfly-react/releases)
- [Commits](https://github.com/patternfly/patternfly-react/compare/@patternfly/react-table@6.2.0...@patternfly/react-table@6.2.2)

---
updated-dependencies:
- dependency-name: "@patternfly/react-table"
  dependency-version: 6.2.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lucferbux committed:
Enable shared library on model registry (#1008)

* Enable shared library on model registry

Signed-off-by: lucferbux <lferrnan@redhat.com>

* Refactor model registry integration and context usage

- Updated package.json to reference local mod-arch-shared module.
- Refactored App component to utilize useModularArchContext for integration checks.
- Removed unused constants and functions related to standalone mode.
- Added new hook useModelRegistriesSettings for fetching model registry settings.
- Updated ModelRegistryRoutes to conditionally render based on integration status.
- Refactored ModelVersionDetailsTabs to use integration context.
- Adjusted FormFieldset and ThemeAwareSearchInput to import theme context from mod-arch-shared.
- Modified DeleteModelRegistryModal to use ModelRegistryKind type.
- Updated ModelRegistriesTable and ModelRegistriesTableRow to work with ModelRegistryKind.
- Removed deprecated convertToK8sResourceCommon function.
- Introduced utility constants for deployment and platform modes.
- Enhanced bootstrap process to include ModularArchContextProvider with configuration.

Signed-off-by: lucferbux <lferrnan@redhat.com>

* Add testing infra

Signed-off-by: lucferbux <lferrnan@redhat.com>

---------

Signed-off-by: lucferbux <lferrnan@redhat.com>

dependabot[bot] committed:
build(deps-dev): bump ruff from 0.11.12 to 0.11.13 in /clients/python (#1188)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.11.12 to 0.11.13.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.12...0.11.13)

---
updated-dependencies:
- dependency-name: ruff
  dependency-version: 0.11.13
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump huggingface-hub from 0.32.3 to 0.32.4 in /clients/python (#1189)

Bumps [huggingface-hub](https://github.com/huggingface/huggingface_hub) from 0.32.3 to 0.32.4.
- [Release notes](https://github.com/huggingface/huggingface_hub/releases)
- [Commits](https://github.com/huggingface/huggingface_hub/compare/v0.32.3...v0.32.4)

---
updated-dependencies:
- dependency-name: huggingface-hub
  dependency-version: 0.32.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump aiohttp from 3.12.6 to 3.12.11 in /clients/python (#1191)

---
updated-dependencies:
- dependency-name: aiohttp
  dependency-version: 3.12.11
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Al-Pragliola committed:
Merge remote-tracking branch 'upstream/main' into al-pragliola-20250609-sync

Al-Pragliola commented:
/test images

dhirajsb commented:
/retest

tarilabs commented:
will address https://github.com/openshift/release/blob/master/ci-operator/config/opendatahub-io/model-registry/opendatahub-io-model-registry-main.yaml separately
